### PR TITLE
Feat/optimize deployment

### PR DESCRIPTION
**Pull request by butchtm**
## Optimize deployment

- feat: optimize deployment

## Changelog

* run validation of catalog items only on items changed/added
* run merging of catalog items only on items changed/added/deleted - leaving unchanged items alone
* add checking of changed/unchanged based on git commit time - requires `git restore-mtime` command
* remove cypress record option to prevent overrun of cypress cloud quotas
* commit of catalog.json and featured.json - since these are necessary for comparison between changes
* commit of generated images for items and multi regions  - since detail images for unchanged items will not rerender them
* add  `site-generated.txt` to track site-generation time

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -1,17 +1,27 @@
-name: Test
+name: Test application
 
 on:
   push:
     branches: [main]
+    paths:
+      - 'src/'
+      # - 'scripts/'
+      # - 'tests/'
```
Comment (tm-ger-chagas) [inline .github/workflows/test.yml:9]: nit: remove comments

**Review by tm-ger-chagas — APPROVED** [review]
lgtm 🚀

File: .github/workflows/test.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (butchtm) [reply .github/workflows/test.yml:9]: I'm leaving them in - I'm disabling pytests for now but will turn them on once everything is in place

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
   pull_request:
     branches: [main]
+    paths:
+      - 'src/'
+      # - 'scripts/'
+      # - 'tests/'
```
[the remark this answers is not in the document]
Comment (butchtm) [reply .github/workflows/test.yml:15]: I'm leaving them in - I'm disabling pytests for now but will turn them on once everything is in place

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -33,6 +43,11 @@ jobs:
       - name: Install dependencies
         run: pnpm i && pnpm cypress install && cp .env.github .env
 
+      # - name: Validate scripts
+      #   run: pytest -c all-pytest.ini
+      #   env:
+      #     GSTORAGE_API_KEY: ${{ secrets.GSTORAGE_API_KEY }}
+
```
[the remark this answers is not in the document]
Comment (butchtm) [reply .github/workflows/test.yml:50]: I'm leaving them in - I'm disabling pytests for now but will turn them on once everything is in place

**Pull request merged**